pkg/network/setup/netstat.go

```diff
@@ -76,20 +76,11 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 		}
 	}
 
-	vmi.Status.Interfaces = []v1.VirtualMachineInstanceNetworkInterface{}
-	for _, network := range vmi.Spec.Networks {
-		podIface, err := c.getPodInterfacefromFileCache(vmi, network.Name)
-		if err != nil {
-			return err
-		}
-
-		ifc := v1.VirtualMachineInstanceNetworkInterface{
-			Name: network.Name,
-			IP:   podIface.PodIP,
-			IPs:  podIface.PodIPs,
-		}
-		vmi.Status.Interfaces = append(vmi.Status.Interfaces, ifc)
+	interfacesStatus, err := c.ifacesStatusFromPod(vmi)
+	if err != nil {
+		return err
 	}
+	vmi.Status.Interfaces = interfacesStatus
```

Comment:
nit: I think you could `vmi.Status.Interfaces, err := c.ifacesStatusFromPod(vmi) ....` and make this simpler, but I won't insist.

Reply:
Yes, this is possible but it has a nasty side effect... You get an error and in parallel mutate data. If the caller wants to use this data again, you just corrupted it. Usually, on error, the returned value is undefined.